### PR TITLE
go.mod: github.com/containerd/containerd/v2 v2.0.0-rc.1

### PR DESCRIPTION
**Pull request by AkihiroSuda**
- github.com/containerd/containerd/{containers,images,...} -> github.com/containerd/containerd/v2/core/{containers,images,...}
- github.com/containerd/containerd/{errdefs,logs,platforms} -> github.com/containerd/{errdefs,logs,platforms}
- github.com/containerd/typeurl -> github.com/containerd/typeurl/v2
- github.com/urfave/cli -> github.com/urfave/cli/v2 (See containerd/containerd#9809)

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
```
--- FAIL: TestInit (0.00s)
    payload_test.go:28: failed to detect gogo: false
FAIL
FAIL	github.com/containerd/imgcrypt/images/encryption	0.006s
```
https://github.com/containerd/imgcrypt/actions/runs/8448335008/job/23140237920?pr=148

File: images/encryption/payload.go
Comment (AkihiroSuda) [inline images/encryption/payload.go:1]: I removed this file, but not sure if this is correct

**Comment by AkihiroSuda** [comment]
```
ctr: unhandled media type application/vnd.in-toto+json: invalid argument
Error: Image layerinfo on plain image failed
```

https://github.com/containerd/imgcrypt/actions/runs/8451047452/job/23148583256?pr=148

**Comment by AkihiroSuda** [comment]
```
Testing creation of container from encrypted image
Encrypting docker.io/library/bash:latest to docker.io/library/bash:enc
ctr: content digest sha256:4f02cf678855fbab38b3638a131fb83cbbea5e1f815dce93c2d8a9d12796e541: not found
Error: Image encryption with JWE failed; public key: jwe:/tmp/tmp.HZ76rjzgUp/mypubkey.jwk
```

https://github.com/containerd/imgcrypt/actions/runs/8453433526/job/23156116537?pr=148

**Comment by yylt** [comment]
the failed about CI test may related to the #69

/cc @stefanberger

**Comment by dmcgowan** [comment]
Carried in #157 with the tests fixed

**Pull request closed without merging**